### PR TITLE
[WIP] UDockerTask + CARETask refactor

### PR DESCRIPTION
**Pull request by jopasserat**
Will solve the issue with retrieving the results from output folder in the UDockerTask
  - [X] factor resolvers out to new container module
  - [X] update CARETask
  - [X] update UDockerTask
  - [X] beautify
  - [X] `sharedContainers` can lead to race conditions (_kind of solved_ - see 8594999f5420ee37b9bcdfa029eddd82201b4bba)
  - [ ] reuse OpenMOLE storage mechanisms when pulling image layers

### CODE REVIEW

File: openmole/plugins/org.openmole.plugin.task.care/src/main/scala/org/openmole/plugin/task/care/CARETask.scala [hunk 1]
```diff
@@ -185,7 +185,7 @@ object CARETask extends Logger {
     def rootDirectory = extractedArchive / rootfs
 
     def outputPathResolver(filePath: String): File = {
-      def isParent(dir: String, file: String) = File(file).getAbsolutePath.startsWith(File(dir).getAbsolutePath)
+      def isParent(dir: String, file: String) = dir.equals(File(file).getParent)
```
Comment (romainreuillon) [inline openmole/plugins/org.openmole.plugin.task.care/src/main/scala/org/openmole/plugin/task/care/CARETask.scala]: This is not equivalent, what about non level 1 subdirs ?

Comment (romainreuillon) [inline openmole/plugins/org.openmole.plugin.task.care/src/main/scala/org/openmole/plugin/task/care/CARETask.scala]: => Bug

**Comment by jopasserat** [comment]
OK done with review / refactor :smile:

**Comment by romainreuillon** [comment]
Merged

**Pull request closed without merging**